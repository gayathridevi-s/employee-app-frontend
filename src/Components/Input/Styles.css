.input {
    width: 460px;
    height: 50px;
    border: 1px solid #B4B7BD;
    border-radius: 5px;
    opacity: 1;
}