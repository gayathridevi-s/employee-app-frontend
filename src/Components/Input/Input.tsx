import { FC } from 'react';
import './Styles.css';

export type InputPropsTypes = {
  value: string|number;
  onChange: (e) => void;
  label: string;
  placeholder: string;
  type: 'text' | 'password' | 'number';
};
const Input: FC<InputPropsTypes> = (props) => {
  return (
    <div>
      <div>{props.label}</div>
      <input
        className='input'
        type={props.type}
        value={props.value}
        placeholder={props.placeholder}
        onChange={(e) => props.onChange(e.target.value)}
      />
    </div>
  );
};

export default Input;
