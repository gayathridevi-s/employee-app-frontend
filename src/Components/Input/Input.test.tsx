import { render, screen, fireEvent } from '@testing-library/react';
import Input, { InputPropsTypes } from './Input';

describe('Input Component', () => {
  test('renders label and placeholder correctly', () => {
    const props: InputPropsTypes = {
      value: '',
      onChange: () => {},
      label: 'Username',
      placeholder: 'Username',
      type: 'text'
    };

    render(<Input {...props} />);

    const labelElement = screen.getByText('Username');
    const inputElement = screen.getByPlaceholderText('Username');

    expect(labelElement).toBeTruthy();
    expect(inputElement).toBeTruthy();
  });

  test('updates value on input change', () => {
    let inputValue = '';
    const handleChange = (newValue) => {
      inputValue = newValue;
    };

    const props: InputPropsTypes = {
      value: inputValue,
      onChange: handleChange,
      label: 'Username',
      placeholder: 'Username',
      type: 'text'
    };

    render(<Input {...props} />);

    const inputElement = screen.getByPlaceholderText('Username');

    fireEvent.change(inputElement, { target: { value: 'newusername' } });

    expect(inputValue).toBe('newusername');
  });

  test('matches snapshot', () => {
    const props: InputPropsTypes = {
      value: '',
      onChange: () => {},
      label: 'Username',
      placeholder: 'Enter your username',
      type: 'text'
    };

    const { asFragment } = render(<Input {...props} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
