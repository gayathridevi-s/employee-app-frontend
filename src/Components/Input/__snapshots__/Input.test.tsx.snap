// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      Username
    </div>
    <input
      class="input"
      placeholder="Enter your username"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;
